File: Question4/Question4.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

function solution(n, employees, shifts) {
    // Write solution here
}

function main() {
    const firstLine = readLine().split(' ');

    const n = parseInt(firstLine[0]);

    const employees = readLine().split(' ');

    let shifts = [];
    for (let i = 0; i < n; i++) {
        const temp = readLine().split(' ');
        shifts.push([parseInt(temp[0]), parseInt(temp[1])]);
    }

    const answer = solution(n, employees, shifts);
}

File: Question4/Question4.java
import java.io.*;
import java.util.*;

class Solution {
    static void solution(int n, List<String> employees, List<List<Integer>> shifts) {
        // Write solution here
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        List<String> employees = new ArrayList<String>();
        for (int i = 0; i < n; i++) {
            employees.add(br.readLine());
        }

        List<List<Integer>> shifts = new ArrayList<List<Integer>>();
        for (int i = 0; i < n; i++) {
            List<Integer> currentShift = new ArrayList<Integer>(2);
            currentShift.add(Integer.parseInt(br.readLine()));
            currentShift.add(Integer.parseInt(br.readLine()));
            shifts.add(currentShift);
        }

        solution(n, employees, shifts);
    }
}
